src/core/ddsi/src/ddsi_security_omg.c

```diff
@@ -2334,73 +2441,83 @@ static bool q_omg_security_register_remote_reader_match(struct proxy_reader *prd
   struct ddsi_domaingv *gv = pp->e.gv;
   struct dds_security_context *sc = q_omg_security_get_secure_context(pp);
   DDS_Security_SecurityException exception = DDS_SECURITY_EXCEPTION_INIT;
-  struct proxypp_pp_match *pm;
-  struct security_entity_match *match;
+  struct proxypp_pp_match *proxypp_match;
+  struct wr_prd_match *match;
   bool send_tokens = false;
+  bool allowed = false;
 
-  *crypto_handle = 0;
-
-  if ((pm = get_pp_proxypp_match_if_authenticated(pp, proxypp, prd->e.guid.entityid)) == NULL)
+  if ((proxypp_match = get_pp_proxypp_match_if_authenticated(pp, proxypp, prd->e.guid.entityid)) == NULL)
     return false;
 
   /* TODO: the security_entity_match should be removed after the the received tokens are stored in the plugin.
```

Comment:
Given that this "TODO" disappeared in ``q_omg_security_register_remote_writer_match`` I'm inclined to think it was meant to be removed here as well.